tests/store/test_datasource.py

```diff
@@ -550,13 +616,38 @@ def test_add_data_with_overlap_check_stored_dataset(bucket, datasets_with_overla
     data_b = xr.Dataset({"mf": ("time", values_b)}, coords={"time": time_b}, attrs=attributes)
     data_c = xr.Dataset({"mf": ("time", values_c)}, coords={"time": time_c}, attrs=attributes)
 
+    test_hash_a = {"filename_a": "test-hash-123"}
+    test_hash_b = {"filename_b": "test-hash-124"}
+    test_hash_c = {"filename_c": "test-hash-125"}
+
     attributes = create_attributes()
 
     d = Datasource(bucket=bucket)
 
-    d.add_data(metadata=attributes, data=data_a, data_type="surface", new_version=False, if_exists="combine")
-    d.add_data(metadata=attributes, data=data_b, data_type="surface", new_version=False, if_exists="combine")
-    d.add_data(metadata=attributes, data=data_c, data_type="surface", new_version=False, if_exists="combine")
+    d.add_data(
+        metadata=attributes,
+        data=data_a,
+        file_hashes=test_hash_a,
+        data_type="surface",
+        new_version=False,
+        if_exists="combine",
```

Comment:
This may be unrelated to this PR but at the moment the `if_exists="combine"` mode isn't doing anything so any ideas what this is doing here?

Reply:
These are marked xfail, I just updated the tests so they could be used when combine is valid again.